lib/krb5/cache.c

```diff
@@ -783,8 +839,14 @@ krb5_cc_configured_default_name(krb5_context context)
         return context->configured_default_cc_name;
 
 #ifdef _WIN32
-    if ((expanded = _krb5_get_default_cc_name_from_registry(context)))
-        return context->configured_default_cc_name = expanded;
+    if ((expanded = _krb5_get_default_cc_name_from_registry(context))) {
+        ret = resolve_cfg_default_name(context, expanded,
+                                       &context->configured_default_cc_name);
+        if (ret == 0) {
+            free(expanded);
+            return context->configured_default_cc_name;
```

Comment:
`context->configured_default_cc_name` is assigned by the function but should it return the pointer or a copy?

`krb5_cccol_get_default_ccname()` for example returns a copy.

When `krb5_cc_configured_default_name()` is called from `krb5_cc_set_default_name()` a copy is immediately made of the return value.

Reply:
The function returns `context->configured_default_cc_name` as a `const char *`.  We have this pattern elsewhere nearby.